### DISCUSSION TITLE
questions regarding redis-rdb-cli

### DISCUSSION THREAD

**Discussion by bluecrabs007** [leonchen83/redis-cli-tool#30] [General]
Hi,
Thank you for such a wonderful tool!
I have a question.
Is there a different method to back up a redis cluster other than iterating over each master with 
```rdt -b redis://$eachMaster:6379 -o /path/to/$eachMasterDump.rdb```
https://github.com/leonchen83/redis-rdb-cli#backup-remote-rdb

**Comment by leonchen83** [comment]
Hi, 
you are right. for now all redis-rdb-cli commands does not support for cluster from `--source`. but target support for cluster.
for example
`rmt -s redis://host:port -m redis://clusterhost:clusterport -r`

**Comment by bluecrabs007** [comment]
Thanks @leonchen83 for your prompt response.
I have another question.
Is there a way to restore specific slots from a rdb into a remote redis cluster?

**Comment by leonchen83** [comment]
Hi
@bluecrabs007 
there is an indirect method to do this.
1. create a standard cluster `/path/to/nodes.conf`
```
126be0dd065a87eea5cb4036d1d4ee958c02078a 127.0.0.1:10002@20002 master - 0 1531044047088 2 connected 1234
34b6e1dfb871ad30398ef5edd6b9a954617e6ec1 127.0.0.1:10003@20003 master - 0 1531044047088 3 connected 1235-16383
89d020a7e727e81f003836207902ae26fe05fd51 127.0.0.1:10001@20001 myself,master - 0 1531044047000 1 connected 0-1233
vars currentEpoch 6 lastVoteEpoch 0
```
2. split rdb file base on `nodes.conf`
```
$ rdt -s /path/to/dump.rdb -c /path/to/nodes.conf -o /path/to
```
3. migrate `/path/to/126be0dd065a87eea5cb4036d1d4ee958c02078a.rdb` to your target cluster
```
$ rmt -s /path/to/126be0dd065a87eea5cb4036d1d4ee958c02078a.rdb -m redis://clusterhost:clusterport -r
```

`126be0dd065a87eea5cb4036d1d4ee958c02078a 127.0.0.1:10002@20002 master - 0 1531044047088 2 connected 1234` means only assign slot `1234` to this node. rdt can split rdb base this `nodes.conf`
